src/graphql/mirror.js

```diff
@@ -886,3 +886,54 @@ function _primitivesTableName(typename: Schema.Typename) {
   }
   return `"primitives_${typename}"`;
 }
+
+/**
+ * Convert a prepared statement into a JS function that executes that
+ * statement and asserts that it makes exactly one change to the
+ * database.
+ *
+ * The prepared statement must use only named parameters, not positional
+ * parameters.
+ *
+ * The prepared statement must not return data (e.g., INSERT and UPDATE
+ * are okay; SELECT is not).
+ *
+ * The statement is not executed inside an additional transaction, so in
+ * the case that the assertion fails, the effects of the statement are
+ * not rolled back by this function.
+ *
+ * This is useful when the statement is like `UPDATE ... WHERE id = ?`
+ * and it is assumed that `id` is a primary key for a record already
+ * exists---if either existence or uniqueness fails, this method will
```

Comment:
a: When writing in a plain-text, non–user facing format (primarily code
comments), I generally try to stick to ASCII when reasonable. This
primarily means omitting smart quotes, smart ellipses, and correct
dashes and minus signs. I do this primarily for the benefit of people
who would have a harder time editing the nicer forms, perhaps because
they get confused by a diff of `-` to `−`, or they don’t know how to
input `df…` in Vim, or they’ve SSHed into an old system that doesn’t
have good Unicode support.

(Within ASCII, the choices are usually easy: `---` is the standard
ligature for “—” since TeX days, `...` is obvious for “…”, etc.)

I am not dogmatic about this. For example, our `config/env.js` file
contains the line

```
  // Useful for determining whether we’re running in production mode.
```

which I have not excised, and I certainly don’t correct others on this
regard.